### DISCUSSION TITLE
Can't get seaweedfs running, connection refused on grpc port, following the guides

### DISCUSSION THREAD

**Discussion by kpko** [seaweedfs/seaweedfs#5866] [Q&A]
I was trying seaweed as an alternative to minio and I can't get a cluster running on my machines. I created three test vms that are on the same network. They are basic debian VMs, latest version, no firewall.

I was looking at https://github.com/seaweedfs/seaweedfs/wiki/Getting-Started and https://github.com/seaweedfs/seaweedfs/wiki/Failover-Master-Server to see how I can get a simple test setup with my three virtual machines.

I tried running the following command on all three vms (vm01, vm02, vm03):

` ./weed server -master.ip=$(hostname) -master.port=9333 -dir=/data/weed -volume.port=8080 -master.peers=vm01:9333,vm02:9333,vm03:9333`

First of all, seaweedfs errors because master.ip is not a valid option, even though its stated in the documentation. Looking through the --help text I came across the "-ip" option and tried this.

` ./weed server -ip=$(hostname) -master.port=9333 -dir=/data/weed -volume.port=8080 -master.peers=vm01:9333,vm02:9333,vm03:9333`

Using that command, every server tries to connect to the other servers but they fail:

```
kp@vm01:~$ ./weed server -ip=$(hostname) -master.port=9333 -dir=/data/weed -volume.port=8080 -master.peers=vm01:9333,vm02:9333,vm03:9333
I0806 18:46:26.131757 master.go:280 current: vm01:9333 peers:vm01:9333,vm02:9333,vm03:9333
I0806 18:46:26.131891 file_util.go:27 Folder /data/weed Permission: -rwxr-xr-x
I0806 18:46:26.131949 master.go:280 current: vm01:9333 peers:vm01:9333,vm02:9333,vm03:9333
I0806 18:46:26.131974 file_util.go:27 Folder /data/weed Permission: -rwxr-xr-x
I0806 18:46:26.132052 master_server.go:133 Volume Size Limit is 30000 MB
I0806 18:46:26.132192 master.go:161 Start Seaweed Master 30GB 3.71 ed7e721efe82a29b031e39e37b729a536e6cde04 at vm01:9333
I0806 18:46:26.132442 volume_grpc_client_to_master.go:43 checkWithMaster vm01:9333: get master vm01:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 127.0.1.1:19333: connect: connection refused"
I0806 18:46:26.132444 raft_server.go:118 Starting RaftServer with vm01:9333
I0806 18:46:26.134252 volume_grpc_client_to_master.go:43 checkWithMaster vm02:9333: get master vm02:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.188.143:19333: connect: connection refused"
I0806 18:46:26.135725 volume_grpc_client_to_master.go:43 checkWithMaster vm03:9333: get master vm03:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.188.107:19333: connect: connection refused"
I0806 18:46:26.139792 raft_server.go:167 current cluster leader:
I0806 18:46:27.926937 volume_grpc_client_to_master.go:43 checkWithMaster vm01:9333: get master vm01:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 127.0.1.1:19333: connect: connection refused"
I0806 18:46:27.928971 volume_grpc_client_to_master.go:43 checkWithMaster vm02:9333: get master vm02:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.188.143:19333: connect: connection refused"
I0806 18:46:27.930540 volume_grpc_client_to_master.go:43 checkWithMaster vm03:9333: get master vm03:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.188.107:19333: connect: connection refused"
```

The ip addresses seem to be correctly resolved by the hostname but there does not seem to be any service running on 19333. I tested if the port is blocked by something else via "nc -lvp 19333" on one machine and "nc vm02 19333" on the other one but this works flawlessly. 

Could somebody give me a heads up for what I am doing wrong? Thank you very much!

**Comment by chrislusf** [comment]
can not tell. Maybe wait a bit longer.
